src/lib/zod/courses.ts

```diff
@@ -42,6 +57,10 @@ const courseSchemaBase = z
           message: 'Start date cannot be in the past',
         }
       ),
+    lastEnrollDate: z.preprocess(
+      (value) => (value === '' || value === undefined ? null : value),
```

Comment:
This could probably be simplified with 
```suggestion
      (value) => (!value ? null : value),
```

Reply:
Very true, I'l change it!